contracts/vaults/BaseVaults/RibbonThetaVault.sol

```diff
@@ -297,6 +297,17 @@ contract RibbonThetaVault is RibbonVault, RibbonThetaVaultStorage {
         transferAsset(msg.sender, amount);
     }
 
+    /**
+     * @notice Initiates a withdrawal that can be processed once the round completes
+     * @param numShares is the number of shares to withdraw
+     */
+    function initiateWithdraw(uint256 numShares) external nonReentrant {
+        _initiateWithdraw(numShares);
+        currentQueuedWithdrawShares = currentQueuedWithdrawShares.add(
```

Comment:
CEI pattern

Reply:
checks effects interactions doesn't apply here since there arent any interactions, there's only checks and effects - no external contracts get called

considering removing `nonReentrant`